pkg/volume/cephfs/cephfs.go

```diff
@@ -397,12 +377,43 @@ func (cephfsVolume *cephfs) execFuseMount(mountpoint string) error {
 	}
 	src += hosts[i]
 
+	configArgs := []string{}
+	configArgs = append(configArgs, fmt.Sprintf("[client.%s]", cephfsVolume.id))
+	configArgs = append(configArgs, fmt.Sprintf("mon_host = %s", src))
+	configArgs = append(configArgs, fmt.Sprintf("keyring = %s", keyring_file))
+	configArgs = append(configArgs, "client_reconnect_stale = true")
```

Comment:
This is not usually desirable. If the mount becomes stale and gets evicted, none of its in-flight I/O is valid or consistent with the other clients. I would assume the pod should be restarted and the application restarted.

Reply:
Once we meet some network issue or mds restarted, all client can be stale. And if not reconnect it, `umont -f` is needed.

Reply:
> Once we meet some network issue or mds restarted, all client can be stale. And if not reconnect it, umont -f is needed.

Client mounts do not usually get disconnected by the MDS restarting. A client side network issue may result in some mounts failing but even that should not generally cause all clients to become disconnected.

> We often run a task with 30000+ container mounted cephfs, it is nightmare to handle this manually.

I would suggest that restarting the pods should be automated.

In any case, if you are adamant about this config option to solve your problem , I would ask that it remain off-by-default and that it be turned on explicitly somehow.

Reply:
Will split it to another pr.